deltacat/tests/local_deltacat_storage/__init__.py

```diff
@@ -820,33 +820,39 @@ def commit_partition(
         ).all_items()
         or []
     )
+
     partition_deltas: Optional[List[Delta]] = (
         list_partition_deltas(
             partition, ascending_order=False, *args, **kwargs
         ).all_items()
         or []
     )
-    previous_partition_deltas_spos_gt: List[Delta] = [
-        delta
-        for delta in previous_partition_deltas
-        if delta.stream_position > partition_deltas[0].stream_position
-    ]
-    # handle the case if the previous partition deltas have a greater stream position than the partition_delta
-    partition_deltas = previous_partition_deltas_spos_gt + partition_deltas
+
```

Comment:
I would still prefer separate private method for merge_and_promote for ease of understanding the code.

Reply:
sure why not